lib/components/overlays/quick_settings/quick_settings_overlay.dart

```diff
@@ -96,12 +96,20 @@ class _QuickSettingsOverlayState extends State<QuickSettingsOverlay>
               child: Padding(
                 padding: const EdgeInsets.all(16.0),
                 child: MaterialApp(
-                  routes: {
-                    "/": (context) => const QsMain(),
-                    "/pages/account": (context) => const QsAccountPage(),
-                    "/pages/network": (context) => const QsNetworkPage(),
-                    "/pages/theme": (context) => const QsThemePage(),
-                    "/pages/language": (context) => const QsLanguagePage(),
+                  onGenerateInitialRoutes: (initialRoute) => [
+                    customRouteTransition(const QsMain()),
+                  ],
+                  onGenerateRoute: (settings) {
+                    switch (settings.name) {
+                      case '/pages/account':
+                        return customRouteTransition(const QsAccountPage());
+                      case '/pages/network':
+                        return customRouteTransition(const QsNetworkPage());
+                      case '/pages/theme':
+                        return customRouteTransition(const QsThemePage());
+                      case '/pages/language':
+                        return customRouteTransition(const QsLanguagePage());
+                    }
```

Comment:
Consider just returning customRouteaTransition and for its parameter use a switch expression returning the various pages, if needed i can provide an example later on

Reply:
Yeah, I'm not quite sure what you mean. We return the same animation for every given route, depending only on its name which is passed down. If we use a switch expression, we'd need to return `PageRouteBuilder` a bunch of times inside the function.

Reply:
Also personally, I like being able to see the routes right next to the `MaterialApp` widget instead of having them nested in a function somewhere.

Reply:
```dart
return customRouteTransition(
  switch(settings.name) {
    '/pages/account' => const QsAccountPage(),
    '/pages/network' => const QsNetworkPage(),
    '/pages/theme' => const QsThemePage(),
    '/pages/language' => const QsLanguagePage(),
  }
);
```

here, this syntax was introduced with dart 3.0

Reply:
Ahhhhh ye I see. :ok_hand:

Reply:
It's kinda ugly cause I had to add cases for `String()` and `null` or the analyser would complain...

```
  return customRouteTransition(
    switch (settings.name) {
      '/pages/account' => const QsAccountPage(),
      '/pages/network' => const QsNetworkPage(),
      '/pages/theme' => const QsThemePage(),
      '/pages/language' => const QsLanguagePage(),
      String() => const QsMain(),
      null => const QsMain(),
    },
  );
},
```

Wdyt?

Reply:
oops my bad, just use `_ => const QsMain()` instead of String() and null

Reply:
Man my dart knowledge is really rusty xD